### PR TITLE
[wip] new resource - health - #679

### PR DESCRIPTION
**Pull request by joshuaroot**
* adding support for health resource (aws health)

### CODE REVIEW

**Comment by kapilt** [comment]
health events are interesting, as a standalone resource they will require a query filter due to the amount of historical data. there are effectively two categories of health events, account level events, which a separate health event resource is the appropriate abstraction, and then secondarily resource level health events, for which a filter is appropriate (ec2, ebs, etc). we'll likely want both, but keeping this pr bounded/scoped to the health resource, sounds good. also you should talk to MichaelJ as he had was spending some cycles looking at this.

**Comment by joshuaroot** [comment]
I actually thought about trying to make it a query resource... because of the amount of data, but thought I'd start at the basic and move up.

I actually started this because I had talked to MichaelJ on Tuesday to go over some health related items... which led me to try to make a resource work (as it did not exist). Noticed the issue after.

File: c7n/resources/health.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from botocore.exceptions import ClientError
+
+from c7n.actions import BaseAction, ActionRegistry
+from c7n.filters import ValueFilter, Filter, OPERATORS, FilterRegistry
+from c7n.manager import resources
+from c7n.query import ResourceManager
+from c7n.utils import local_session, type_schema
+
+
+filters = FilterRegistry('aws.health.actions')
```
Comment (kapilt) [inline c7n/resources/health.py]: these are not needed, they get gen'd automatically by the resource metaclass

File: c7n/resources/health.py [hunk 2]
```diff
@@ -11,41 +11,121 @@
...
+        dimension = None
+        date = 'startTime'
+        config_type = 'AWS::Health::Event'
+
+    filter_registry = filters
+    action_registry = actions
+
+    def __init__(self, ctx, data):
+        super(HealthEvents, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(self.data.get('query', []))
```
Comment (kapilt) [inline c7n/resources/health.py]: this needs a sensible default, we can't default to querying with no params, when they keep effectively lifetime historical that i've seen.

**Comment by joshuaroot** [comment]
accidentally deleted branch. restored & reopen pr

**Comment by kapilt** [comment]
seems like other folks are building on this now, i think we'll probably need an augment to get the details and maybe a default service filter on the query, i'd like to keep those with resource specific health info out of the default queries for this as a standalone resource. just merging as is, and let's try to evolve from there.

**Pull request merged**